### ISSUE TITLE
(DynamicBungeeAuth now fixed) SCAutoBroadcaster unresolved issues

### ISSUE THREAD

**Issue by Gyztor** [A248/LibertyBans#152]
### LibertyBans Version

LibertyBans 1.0.1

### I have confirmed that ...

- [x] LibertyBans is up to date
- [X] No similar issue has been reported

### Platform

Spigot/Paper, Bungeecord

### Description

There are currently two errors that i usually see appear one at startup and one when someone joins the server
One at startup with plugins SCAutoBroadcaster, DynamicBungeeAuth(Eventually for bedrock people):
```
[02:40:47 WARN] [LibertyBans]: *******************************************
<Unknown plugin> has a critical bug. That plugin has shaded the library 'HikariCP' but did not relocate it, which will pose problems. 
LibertyBans is not guaranteed to function if you do not fix this bug in <Unknown plugin>
Contact the author of this plugin and tell them to relocate their dependencies. Unrelocated class detected was com.zaxxer.hikari.HikariConfig
Note for advanced users: Understanding the consequences, you can disable this check by setting the system property libertybans.relocationbug.disablecheck to 'true'
*******************************************
[02:40:47 WARN] [LibertyBans]: *******************************************
<Unknown plugin> has a critical bug. That plugin has shaded the library 'Kyori-Adventure' but did not relocate it, which will pose problems. 
LibertyBans is not guaranteed to function if you do not fix this bug in <Unknown plugin>
Contact the author of this plugin and tell them to relocate their dependencies. Unrelocated class detected was net.kyori.adventure.audience.Audience
Note for advanced users: Understanding the consequences, you can disable this check by setting the system property libertybans.relocationbug.disablecheck to 'true'
*******************************************
[02:40:47 WARN] [LibertyBans]: *******************************************
<Unknown plugin> has a critical bug. That plugin has shaded the library 'Kyori-Examination' but did not relocate it, which will pose problems. 
LibertyBans is not guaranteed to function if you do not fix this bug in <Unknown plugin>
Contact the author of this plugin and tell them to relocate their dependencies. Unrelocated class detected was net.kyori.examination.Examinable
Note for advanced users: Understanding the consequences, you can disable this check by setting the system property libertybans.relocationbug.disablecheck to 'true'
*******************************************
[02:40:48 INFO] [space.arim.libertybans.core.AbstractBaseFoundation]: Conducting START...
[02:40:48 INFO] [space.arim.libertybans.core.protectedlibs.com.zaxxer.hikari.HikariDataSource]: LibertyBansTemporaryPool-MariaDB - Starting...
[02:40:48 INFO] [space.arim.libertybans.core.protectedlibs.com.zaxxer.hikari.pool.HikariPool]: LibertyBansTemporaryPool-MariaDB - Added connection org.mariadb.jdbc.MariaDbConnection@4d437408
[02:40:48 INFO] [space.arim.libertybans.core.protectedlibs.com.zaxxer.hikari.HikariDataSource]: LibertyBansTemporaryPool-MariaDB - Start completed.
[02:40:48 INFO] [space.arim.libertybans.core.protectedlibs.com.zaxxer.hikari.HikariDataSource]: LibertyBansTemporaryPool-MariaDB - Shutdown initiated...
[02:40:48 INFO] [space.arim.libertybans.core.protectedlibs.com.zaxxer.hikari.HikariDataSource]: LibertyBansTemporaryPool-MariaDB - Shutdown completed.
[02:40:48 INFO] [space.arim.libertybans.core.protectedlibs.com.zaxxer.hikari.HikariDataSource]: LibertyBansPool-MariaDB - Starting...
[02:40:48 INFO] [space.arim.libertybans.core.protectedlibs.com.zaxxer.hikari.pool.HikariPool]: LibertyBansPool-MariaDB - Added connection org.mariadb.jdbc.MariaDbConnection@33299272
[02:40:48 INFO] [space.arim.libertybans.core.protectedlibs.com.zaxxer.hikari.HikariDataSource]: LibertyBansPool-MariaDB - Start completed.
[02:40:48 INFO] [org.flywaydb.core.internal.license.VersionPrinter]: Flyway Community Edition 8.2.0 by Redgate
[02:40:48 INFO] [org.flywaydb.core.internal.database.base.BaseDatabaseType]: Database: jdbc:mariadb://someiphere:3306/libertybans (MariaDB 10.7)
[02:40:48 WARN] [org.flywaydb.core.internal.database.base.Database]: Flyway upgrade recommended: MariaDB 10.7 is newer than this version of Flyway and support has not been tested. The latest supported version of MariaDB is 10.6.
[02:40:48 INFO] [org.flywaydb.core.internal.command.DbValidate]: Successfully validated 6 migrations (execution time 00:00.025s)
[02:40:48 INFO] [org.flywaydb.core.internal.command.DbMigrate]: Current version of schema `libertybans`: 16
[02:40:48 INFO] [org.flywaydb.core.internal.command.DbMigrate]: Schema `libertybans` is up to date. No migration necessary.
```
The kyori error comes from SCAutoBroadcaster, i have asked the dev of that plugin to update kyori to the latest and this error still appears, i have been trying to get in contact with the creator of DynamicBungeeAuth but have not had any response on their part. So far it has not caused any extreme errors other than this but can be easily replicated by installing SCAutoBroadcaster (https://www.spigotmc.org/resources/scautobroadcaster.66378/), and DynamicBungeeAuth (it's a premium plugin but will still post link for reference points: https://www.spigotmc.org/resources/dynamicbungeeauth-premium-command-semi-premium-system-sessions.27480/)
this is what the developer of SCAutoBroadcast said on it (and as mentioned above we tried updating his kyori api verison):
![image](https://user-images.githubusercontent.com/44737827/170935199-05ae60c6-2f9b-4f21-aad4-d260a573949d.png)

**Comment by KoxSosen** [comment]
Hello!

Apologies about the late reply. 

Basically, the reason why you're seeing these errors, is becuase in your server there are plugins, that did not relocate their shaded dependencies.

Libertybans uses it's own mehanism for including dependencies, so we can use different dependencies depending on the platform. LibertyBans uses the platform's adventure integration if it exits, and fails back to a different mehanism if it doesn't. Because of this, Adventure can't be relocated in in LibertyBans, and therefore we need another mehanism in order to use adventure, but not expose it to other plugins unrelocated; An isloated classloader.  

To sum up, the previusly mentioned plugins will have to relocate their shaded dependencies. An example for maven can be found [here](https://github.com/A248/AnvilPowered-relocated/blob/d0681ae50cc4b2442a8d69c06d855853a9640783/pom.xml#L43).

**Comment by A248** [comment]
Hello, thanks for the report. We have a lot of experience handling this issue as it has happened before. I can see there is some confusion as to the cause of the error. Here are some points of note.

First, this issue isn't related to the version of the Kyori library, whether in LibertyBans or SCAutoBroadcaster. As such, updating the library version in SCAutoBroadcaster or LibertyBans won't solve the problem.

Secondly, this matter affects many plugins, not just LibertyBans. However, it often especially affects LibertyBans because we rely on some rarely-used programming mechanisms. So, we print a warning indicating this problem and suggesting how to solve it. Although this is technically a warning, it is very likely to lead to errors, so we encourage it to be solved as soon as possible.

Thirdly, the bug has to be solved by SCAutoBroadcaster and DynamicBungeeAuth. The Kyori library message comes from SCAutoBroadcaster. The message about HikariCP, another library, almost certainly comes from DynamicBungeeAuth. I know this despite that DynamicBungeeAuth is premium, because DynamicBungeeAuth relies on MySQL/MariaDB, and HikariCP is a database-related library.

Fourthly, if necessary, I am able to offer assistance in resolving this bug by working with the developers of SCAutoBroadcaster or DynamicBungeeAuth. Specifically, the bug is that the libraries in question are not *relocated*, meaning their packages are prone to conflict with other plugins. This is, unfortunately, a fairly common bug in the Minecraft plugin ecosystem. However, it is also (usually) very easy to solve; in some rare cases the solution requires significant effort.

**Comment by A248** [comment]
About DynamicBungeeAuth, we reported this bug to them previously:

https://www.spigotmc.org/threads/dynamicbungeeauth-premium-command-semi-premium-system-sessions-paid.171318/page-46#post-4387238

**Comment by Gyztor** [comment]
Sorry for my late reply i misplaced this tab and forgot to check in on it again and just remembered now to check

Let me go contact the SCAutoBroadcaster plugin dev real quick and inform them, i have also been trying to get in contact with the dev of DynamicBungee auth but i have been running on a few weeks of them not verifying me/my purchase. but i do have a few questions too:
1. Can these errors cause anything fatal like memory leaks or something of the sort that could be detrimental or are they in some way ignore-able for the time being? (as i have noticed no other errors but these at bootup)
2. Could these errors cause corruption or un-usability if ignore for now?

i really hope i can get in contact with the DBA dev and start seeing if i can get this fixed on their end because these errors are annoying and libertybans is somewhat the best plugin out there that easily replaces dkbans in useability and while being a little less complicated.

**Comment by Gyztor** [comment]
i have also been very suddenly getting this error on startup it was working before and this wasn't appearing but it seems to be now, unsure if its related but i am putting it here to be safe
this happened when i updated my bungee to the latest than downgraded it back when everything basically died, everything else now seems to work fine except libertybans. so maybe my 1st question has been answered for me lol.
```
[21:04:07 ERROR] [space.arim.libertybans.core.service.StandardAsynchronicityManager]: Exception during miscellaneous asynchronous computation
java.util.concurrent.CompletionException: org.jooq.exception.DataAccessException: SQL [select `libertybans_simple_history`.`id`, `libertybans_simple_history`.`type`, `libertybans_simple_history`.`victim_type`, `libertybans_simple_history`.`victim_uuid`, `libertybans_simple_history`.`victim_address`, `libertybans_simple_history`.`operator`, `libertybans_simple_history`.`reason`, `libertybans_simple_history`.`scope`, `libertybans_simple_history`.`start`, `libertybans_simple_history`.`end` from `libertybans_simple_history` where ((`libertybans_simple_history`.`victim_type` = ? and `libertybans_simple_history`.`victim_uuid` = ?) or (`libertybans_simple_history`.`victim_type` = ? and `libertybans_simple_history`.`victim_uuid` = ?)) order by `libertybans_simple_history`.`start` desc, `libertybans_simple_history`.`id` desc limit ?]; (conn=200430) Prepared statement needs to be re-prepared
        at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315) ~[?:?]
        at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320) ~[?:?]
        at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1770) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
        at java.lang.Thread.run(Thread.java:833) [?:?]
Caused by: org.jooq.exception.DataAccessException: SQL [select `libertybans_simple_history`.`id`, `libertybans_simple_history`.`type`, `libertybans_simple_history`.`victim_type`, `libertybans_simple_history`.`victim_uuid`, `libertybans_simple_history`.`victim_address`, `libertybans_simple_history`.`operator`, `libertybans_simple_history`.`reason`, `libertybans_simple_history`.`scope`, `libertybans_simple_history`.`start`, `libertybans_simple_history`.`end` from `libertybans_simple_history` where ((`libertybans_simple_history`.`victim_type` = ? and `libertybans_simple_history`.`victim_uuid` = ?) or (`libertybans_simple_history`.`victim_type` = ? and `libertybans_simple_history`.`victim_uuid` = ?)) order by `libertybans_simple_history`.`start` desc, `libertybans_simple_history`.`id` desc limit ?]; (conn=200430) Prepared statement needs to be re-prepared
        at org.jooq_3.15.1.MARIADB.debug(Unknown Source) ~[?:?]
        at org.jooq.impl.Tools.translate(Tools.java:2979) ~[?:?]
        at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:643) ~[?:?]
        at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:349) ~[?:?]
        at org.jooq.impl.AbstractResultQuery.fetchLazy(AbstractResultQuery.java:295) ~[?:?]
        at org.jooq.impl.AbstractResultQuery.fetchLazyNonAutoClosing(AbstractResultQuery.java:316) ~[?:?]
        at org.jooq.impl.SelectImpl.fetchLazyNonAutoClosing(SelectImpl.java:2866) ~[?:?]
        at org.jooq.impl.ResultQueryTrait.collect(ResultQueryTrait.java:357) ~[?:?]
        at org.jooq.impl.ResultQueryTrait.fetch(ResultQueryTrait.java:1454) ~[?:?]
        at space.arim.libertybans.core.selector.SelectionImpl.lambda$getSpecificPunishments$3(SelectionImpl.java:257) ~[?:?]
        at space.arim.libertybans.core.database.execute.SQLFunction$1.obtain(SQLFunction.java:42) ~[?:?]
        at space.arim.libertybans.core.database.execute.JooqQueryExecutor.obtainUnfailing(JooqQueryExecutor.java:69) ~[?:?]
        at space.arim.libertybans.core.database.execute.JooqQueryExecutor.lambda$query$0(JooqQueryExecutor.java:200) ~[?:?]
        at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768) ~[?:?]
        ... 3 more
Caused by: java.sql.SQLTransientConnectionException: (conn=200430) Prepared statement needs to be re-prepared
        at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.createException(ExceptionFactory.java:79) ~[?:?]
        at org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.create(ExceptionFactory.java:158) ~[?:?]
        at org.mariadb.jdbc.MariaDbStatement.executeExceptionEpilogue(MariaDbStatement.java:266) ~[?:?]
        at org.mariadb.jdbc.ServerSidePreparedStatement.executeInternal(ServerSidePreparedStatement.java:413) ~[?:?]
        at org.mariadb.jdbc.ServerSidePreparedStatement.execute(ServerSidePreparedStatement.java:364) ~[?:?]
        at space.arim.libertybans.core.protectedlibs.com.zaxxer.hikari.pool.ProxyPreparedStatement.execute(ProxyPreparedStatement.java:44) ~[?:?]
        at space.arim.libertybans.core.protectedlibs.com.zaxxer.hikari.pool.HikariProxyPreparedStatement.execute(HikariProxyPreparedStatement.java) ~[?:?]
        at org.jooq.tools.jdbc.DefaultPreparedStatement.execute(DefaultPreparedStatement.java:219) ~[?:?]
        at org.jooq.impl.Tools.executeStatementAndGetFirstResultSet(Tools.java:4131) ~[?:?]
        at org.jooq.impl.AbstractResultQuery.execute(AbstractResultQuery.java:230) ~[?:?]
        at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:335) ~[?:?]
        at org.jooq.impl.AbstractResultQuery.fetchLazy(AbstractResultQuery.java:295) ~[?:?]
        at org.jooq.impl.AbstractResultQuery.fetchLazyNonAutoClosing(AbstractResultQuery.java:316) ~[?:?]
        at org.jooq.impl.SelectImpl.fetchLazyNonAutoClosing(SelectImpl.java:2866) ~[?:?]
        at org.jooq.impl.ResultQueryTrait.collect(ResultQueryTrait.java:357) ~[?:?]
        at org.jooq.impl.ResultQueryTrait.fetch(ResultQueryTrait.java:1454) ~[?:?]
        at space.arim.libertybans.core.selector.SelectionImpl.lambda$getSpecificPunishments$3(SelectionImpl.java:257) ~[?:?]
        at space.arim.libertybans.core.database.execute.SQLFunction$1.obtain(SQLFunction.java:42) ~[?:?]
        at space.arim.libertybans.core.database.execute.JooqQueryExecutor.obtainUnfailing(JooqQueryExecutor.java:69) ~[?:?]
        at space.arim.libertybans.core.database.execute.JooqQueryExecutor.lambda$query$0(JooqQueryExecutor.java:200) ~[?:?]
        at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768) ~[?:?]
        ... 3 more
Caused by: org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException: Prepared statement needs to be re-prepared
        at org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException.of(MariaDbSqlException.java:34) ~[?:?]
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.exceptionWithQuery(AbstractQueryProtocol.java:194) ~[?:?]
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.exceptionWithQuery(AbstractQueryProtocol.java:177) ~[?:?]
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executePreparedQuery(AbstractQueryProtocol.java:1145) ~[?:?]
        at org.mariadb.jdbc.ServerSidePreparedStatement.executeInternal(ServerSidePreparedStatement.java:406) ~[?:?]
        at org.mariadb.jdbc.ServerSidePreparedStatement.execute(ServerSidePreparedStatement.java:364) ~[?:?]
        at space.arim.libertybans.core.protectedlibs.com.zaxxer.hikari.pool.ProxyPreparedStatement.execute(ProxyPreparedStatement.java:44) ~[?:?]
        at space.arim.libertybans.core.protectedlibs.com.zaxxer.hikari.pool.HikariProxyPreparedStatement.execute(HikariProxyPreparedStatement.java) ~[?:?]
        at org.jooq.tools.jdbc.DefaultPreparedStatement.execute(DefaultPreparedStatement.java:219) ~[?:?]
        at org.jooq.impl.Tools.executeStatementAndGetFirstResultSet(Tools.java:4131) ~[?:?]
        at org.jooq.impl.AbstractResultQuery.execute(AbstractResultQuery.java:230) ~[?:?]
        at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:335) ~[?:?]
        at org.jooq.impl.AbstractResultQuery.fetchLazy(AbstractResultQuery.java:295) ~[?:?]
        at org.jooq.impl.AbstractResultQuery.fetchLazyNonAutoClosing(AbstractResultQuery.java:316) ~[?:?]
        at org.jooq.impl.SelectImpl.fetchLazyNonAutoClosing(SelectImpl.java:2866) ~[?:?]
        at org.jooq.impl.ResultQueryTrait.collect(ResultQueryTrait.java:357) ~[?:?]
        at org.jooq.impl.ResultQueryTrait.fetch(ResultQueryTrait.java:1454) ~[?:?]
        at space.arim.libertybans.core.selector.SelectionImpl.lambda$getSpecificPunishments$3(SelectionImpl.java:257) ~[?:?]
        at space.arim.libertybans.core.database.execute.SQLFunction$1.obtain(SQLFunction.java:42) ~[?:?]
        at space.arim.libertybans.core.database.execute.JooqQueryExecutor.obtainUnfailing(JooqQueryExecutor.java:69) ~[?:?]
        at space.arim.libertybans.core.database.execute.JooqQueryExecutor.lambda$query$0(JooqQueryExecutor.java:200) ~[?:?]
        at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768) ~[?:?]
        ... 3 more
Caused by: java.sql.SQLException: Prepared statement needs to be re-prepared
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readErrorPacket(AbstractQueryProtocol.java:1694) ~[?:?]
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readPacket(AbstractQueryProtocol.java:1556) ~[?:?]
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.getResult(AbstractQueryProtocol.java:1519) ~[?:?]
        at org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executePreparedQuery(AbstractQueryProtocol.java:1142) ~[?:?]
        at org.mariadb.jdbc.ServerSidePreparedStatement.executeInternal(ServerSidePreparedStatement.java:406) ~[?:?]
        at org.mariadb.jdbc.ServerSidePreparedStatement.execute(ServerSidePreparedStatement.java:364) ~[?:?]
        at space.arim.libertybans.core.protectedlibs.com.zaxxer.hikari.pool.ProxyPreparedStatement.execute(ProxyPreparedStatement.java:44) ~[?:?]
        at space.arim.libertybans.core.protectedlibs.com.zaxxer.hikari.pool.HikariProxyPreparedStatement.execute(HikariProxyPreparedStatement.java) ~[?:?]
        at org.jooq.tools.jdbc.DefaultPreparedStatement.execute(DefaultPreparedStatement.java:219) ~[?:?]
        at org.jooq.impl.Tools.executeStatementAndGetFirstResultSet(Tools.java:4131) ~[?:?]
        at org.jooq.impl.AbstractResultQuery.execute(AbstractResultQuery.java:230) ~[?:?]
        at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:335) ~[?:?]
        at org.jooq.impl.AbstractResultQuery.fetchLazy(AbstractResultQuery.java:295) ~[?:?]
        at org.jooq.impl.AbstractResultQuery.fetchLazyNonAutoClosing(AbstractResultQuery.java:316) ~[?:?]
        at org.jooq.impl.SelectImpl.fetchLazyNonAutoClosing(SelectImpl.java:2866) ~[?:?]
        at org.jooq.impl.ResultQueryTrait.collect(ResultQueryTrait.java:357) ~[?:?]
        at org.jooq.impl.ResultQueryTrait.fetch(ResultQueryTrait.java:1454) ~[?:?]
        at space.arim.libertybans.core.selector.SelectionImpl.lambda$getSpecificPunishments$3(SelectionImpl.java:257) ~[?:?]
        at space.arim.libertybans.core.database.execute.SQLFunction$1.obtain(SQLFunction.java:42) ~[?:?]
        at space.arim.libertybans.core.database.execute.JooqQueryExecutor.obtainUnfailing(JooqQueryExecutor.java:69) ~[?:?]
        at space.arim.libertybans.core.database.execute.JooqQueryExecutor.lambda$query$0(JooqQueryExecutor.java:200) ~[?:?]
        at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768) ~[?:?]
        ... 3 more```

**Comment by Gyztor** [comment]
Made a gitlab issue for DynamicBungeeAuth: https://gitlab.com/UraharaNZ/dynamicbungeeauth-issues/-/issues/5

**Comment by UraharaNZ** [comment]
Hey,
Thanks for reporting this issue I'll will take a look on this to release asp the fix.

**Comment by A248** [comment]
> Can these errors cause anything fatal like memory leaks or something of the sort that could be detrimental or are they in some way ignore-able for the time being? (as i have noticed no other errors but these at bootup)
> Could these errors cause corruption or un-usability if ignore for now?

I'll consider these questions as the same because they receive with the same answer. The technically correct answer is "there are no guarantees," because libraries may organize their own code any which way, and the arbitrary order of class-loading allows for any combination of new and old code to exist simultaneously. The analogy is building a bridge with new and old concrete at the same time. In some parts of the bridge, old concrete might be used; in other parts, new concrete. One out of twenty bridges made in this way might collapse, but most of them will probably hold up.**

Classloading is a detailed process and plugins, each of which have their own classloader, further complicate the process. This is to say nothing of extra patches to classloading in the plugin ecosystem (There used to be a bug with library loading, which we tracked down and fixed: [Spigot-6749](https://hub.spigotmc.org/jira/browse/SPIGOT-6749)) I have debugged classloading-related bugs in the past: they are always incredibly difficult to identify and track down. This is exactly why we issue this warning, to prevent the rare case where an insurmountable bug arises because of a trivial mistake made by another plugin.

** - *In addition, due to a detail in how database drivers interact with LibertyBans, you will not be able to use development builds of LibertyBans until this is fixed, but you will be able to use release builds subject to the described hazards*.

> i have also been very suddenly getting this error on startup it was working before and this wasn't appearing but it seems to be now, unsure if its related but i am putting it here to be safe

This is a somewhat well-known bug in MariaDB / MySQL. We cannot solve this. Both [MySQL](https://bugs.mysql.com/bug.php?id=42041) and [MariaDB](https://jira.mariadb.org/browse/MDEV-17124) have open issues for this bug in their tracker. A temporary solution is to increase the value of your database's `table_definition_cache`. If you use a hosting provider, you should contact them asking to have this done. If necessary you can link them to the relevant bug reports.

**Comment by UraharaNZ** [comment]
@A248 Error is now fixed you can close this issue, sorry i didn't see your report on spigotmc.

DBA: 10.30B will not show that error with LibertyBans.

**Comment by A248** [comment]
I'm closing this issue now since any remaining incompatibilities have to be fixed by SCAutoBroadcaster, not us.

If SCAutoBroadcaster is a crucial plugin for you to have and you cannot find a replacement, I am able to manually patch the SCAutoBroadcaster plugin jar. I only do this if absolutely necessary and as a pure convenience to users outside my responsibilities.
